### PR TITLE
Cybersixgill darkfeed confidence filter

### PR DESCRIPTION
**Pull request by syed-loginsoft**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Added UI parameter "Sixgill Confidence".  Records not matching confidence level are not ingested in XSOAR

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

### CODE REVIEW

File: Packs/Sixgill-Darkfeed/Integrations/Sixgill_Darkfeed/Sixgill_Darkfeed.yml [hunk 1]
```diff
@@ -11,6 +11,17 @@ configuration:
...
+  display: Sixgill Confidence
+  name: confidence
+  required: false
+  type: 15
+  options:
+  - all - Ingest all IOCs
+  - 90 - IOC exactly fits description
+  - 80 - IOC almost exactly fits description
+  - 70 - IOC mostly fits description
+  - 60 - IOC generally fits description
```
Comment (GuyAfik) [inline Packs/Sixgill-Darkfeed/Integrations/Sixgill_Darkfeed/Sixgill_Darkfeed.yml:24]:
```suggestion
- defaultvalue: 'all'
  display: Sixgill Confidence
  name: confidence
  required: false
  type: 15
  options:
  - all
  - 90
  - 80
  - 70
  - 60
   additionalinfo: 'Ingest all IOCs=all, IOC exactly fits description=90, IOC almost exactly fits description=80, IOC mostly fits description=70, IOC generally fits description=60'
```

perhaps something like this? https://xsoar.pan.dev/docs/integrations/yaml-file#configuration (see additional-info section)

File: Packs/Sixgill-Darkfeed/Integrations/Sixgill_Darkfeed/Sixgill_Darkfeed.py [hunk 2]
```diff
@@ -326,9 +343,12 @@ def main():
         'test-module': test_module_command,
         'sixgill-get-indicators': get_indicators_command
     }
+    confidence = None
+    if "confidence" in demisto.params():
+        confidence = arg_to_number(demisto.params().get("confidence").split(" - ")[0].strip(), "confidence", required=False)
```
Comment (GuyAfik) [inline Packs/Sixgill-Darkfeed/Integrations/Sixgill_Darkfeed/Sixgill_Darkfeed.py:348]:
```suggestion
        confidence = arg_to_number(demisto.params().get("confidence"))
```

you can remove all of this if implementing my comment in the yml and only handle the case where its `all` and not a number.

File: Packs/Sixgill-Darkfeed/ReleaseNotes/2_2_0.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### Sixgill DarkFeed Threat Intelligence
+
+- Added config parameter "Sixgill Confidence"
+- IOCs not matching confidence level are not saved
```
Comment (GuyAfik) [inline Packs/Sixgill-Darkfeed/ReleaseNotes/2_2_0.md:7]:
```suggestion
- Added a the new integration parameter *Sixgill Confidence* to retrieve indicators with a specific confidence."
```

**Review by GuyAfik — CHANGES_REQUESTED** [review]
@syed-loginsoft  nice work, looks good!

Please see my comments and please add a unit-test as the build fails on test-coverage:

https://app.circleci.com/pipelines/github/demisto/content/255920/workflows/af664de0-58b6-4b1d-b823-f4304adcfa9b/jobs/698288

let me know if you need additional help :)

**Comment by GuyAfik** [comment]
@syed-loginsoft waiting for your changes, let me know if you need any help

File: Packs/Sixgill-Darkfeed/Integrations/Sixgill_Darkfeed/Sixgill_Darkfeed.py [hunk 4]
```diff
@@ -326,9 +343,10 @@ def main():
         'test-module': test_module_command,
         'sixgill-get-indicators': get_indicators_command
     }
+    confidence = arg_to_number(demisto.params().get("confidence"), "confidence", required=False)
```
Comment (GuyAfik) [inline Packs/Sixgill-Darkfeed/Integrations/Sixgill_Darkfeed/Sixgill_Darkfeed.py:346]: that is good, however if confidence = 'all', it won't fail on `ValueError`?

Comment (syed-loginsoft) [reply Packs/Sixgill-Darkfeed/Integrations/Sixgill_Darkfeed/Sixgill_Darkfeed.py:346]: I am verifying, will update you.

Comment (syed-loginsoft) [reply Packs/Sixgill-Darkfeed/Integrations/Sixgill_Darkfeed/Sixgill_Darkfeed.py]: Tested and fixed. Please check. Let me know if anything else is required

**Review by GuyAfik — APPROVED** [review]
@syed-loginsoft looks good, very nice work!

**Pull request merged**